docs/performance/remote-filesystem/amazon-s3.md

```diff
@@ -0,0 +1,377 @@
+---
+title: Amazon S3
+level: advanced
+---
+Amazon S3
+=========
+
+Configuring Bolt to use Amazon Simple Storage Service (Amazon S3) requires the
+addition of Flysystem and caching libraries, as well as custom service provider
+code to your project.
+
+1. Set-up, configure, and confirm operational your Amazon S3 bucket
+1. Update `composer.json` in your project root
+1. Creating service providers
+1. Bootstrap your providers
+
+Prerequisites
+-------------
+
+<p class="warning"><strong>Warning:</strong> This documentation, and examples,
+assume you have a fully configured and working Amazon S3 bucket.
+</p>
+
+
+
+### Required Libraries
+----------------------
+
+You need to add the following libraries to your project:
+
+  * `league/flysystem-aws-s3-v3:^1.0`
+  * `league/flysystem-cached-adapter:^1.0`
+
+This can simply be achieved by running:
+
+```bash
+$ composer require league/flysystem-aws-s3-v3:^1.0 league/flysystem-cached-adapter:^1.0
+```
+
+
+### Caching Libraries
+----------------------
+
+While other cache targets are optional, right now this example will be based on
+Redis. Alternatively any PSR-6 can be used, as well as Doctrine Cache.
+
+To use Redis, you will need a configured and running Redis instance, and need
+to install `predis/predis:^1.1`, which can be achieved by running:
+
+
+```bash
+$ composer require predis/predis:^1.1
+```
```

Comment:
The cache needs to be shared between servers, else bad things happen. We recommend using Elasticache for a shared Redis instance, and using the Predis library to connect to it.